src/lib/TelemetryClient.ts

```diff
@@ -349,7 +349,22 @@ class TelemetryClient {
      * then the specified props will be updated.  Otherwise, a new instance will returned. 
```

Comment:
The "telemetryConfig" instance is retrieved asynchronously, often after the "shell ready" event is queued up, so we need to backfill a lot of values here.  I wanted one method to either create this instance, or backfill an existing instance.

If you're question is why are we updating the passed in props, as opposed to doing a shallow copy, no good reason other than it is slightly more efficient.  Given the narrow (private) uses of this function, I don't think this is subject to misuse or confusion, but I can change if you feel strongly about it.